### PR TITLE
HTML-646 merge HtmlFormEntry19Ext into HtmlFormEntry

### PR DESCRIPTION
**Pull request by PawelGutkowski**
https://issues.openmrs.org/browse/HTML-646

I've executed all steps concerning this module described in issue, beside this one:

> Incorporate the messages.propertes into the HFE messages.properties (be sure to properly push the translations into Transifex); remove the HFE 19ext Transifex project (if one exists)

Messages are moved, I've joined OpenMRS on Transifex, but I think I don't have permissions to push translations or delete projects.

### CODE REVIEW

File: api/src/test/java/org/openmrs/module/htmlformentry/IntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,488 @@
+/**
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/htmlformentry/IntegrationTest.java]: Missed this, but for consistency with the existing tests, let's rename this to EncounterProviderAndRoleTagTest

File: api/src/test/resources/org/openmrs/module/htmlformentry/include/IntegrationTests.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (mogoodrich) [inline api/src/test/resources/org/openmrs/module/htmlformentry/include/IntegrationTests.xml]: let's rename this to encounterProviderAndRole.xml as well

File: api/src/main/java/org/openmrs/module/htmlformentry/HtmlFormEntryActivator.java [hunk 3]
```diff
@@ -27,11 +29,15 @@
 
 	@Override
 	public void started() {
+		Context.getService(HtmlFormEntryService.class).addHandler("encounterProviderAndRole", new EncounterProviderAndRoleTagHandler());
```
Comment (PawelGutkowski) [inline api/src/main/java/org/openmrs/module/htmlformentry/HtmlFormEntryActivator.java]: Oh, I see, you even mentioned it in description. Fixed!

File: api/src/main/java/org/openmrs/module/htmlformentry/HtmlFormEntryActivator.java [hunk 4]
```diff
@@ -27,11 +29,15 @@
 
 	@Override
 	public void started() {
+		Context.getService(HtmlFormEntryService.class).addHandler("encounterProviderAndRole", new EncounterProviderAndRoleTagHandler());
 		log.info("Started HTML Form Entry Module");
 	}
 
 	@Override
 	public void stopped() {
+		HtmlFormEntryService hfes = Context.getService(HtmlFormEntryService.class);
```
Comment (PawelGutkowski) [inline api/src/main/java/org/openmrs/module/htmlformentry/HtmlFormEntryActivator.java]: done

File: api/src/main/java/org/openmrs/module/htmlformentry/HtmlFormEntryUtil.java [hunk 5]
```diff
@@ -1967,4 +1973,240 @@ public static String getArchiveDirPath() {
...
+		return providerRole;
+	}
+
+
+	private static Object getProviderRoleById(Integer providerRoleId) {
+
+		// we have to fetch the provider role by reflection, since the provider management module is not a required dependency
+
+		try {
+			Class<?> providerManagementServiceClass = Context.loadClass("org.openmrs.module.providermanagement.api.ProviderManagementService");
```
Comment (PawelGutkowski) [inline api/src/main/java/org/openmrs/module/htmlformentry/HtmlFormEntryUtil.java]: Done

File: api/src/test/java/org/openmrs/module/htmlformentry/IntegrationTest.java (hunk 1 above)
Comment (PawelGutkowski) [inline api/src/test/java/org/openmrs/module/htmlformentry/IntegrationTest.java]: Fixed

**Comment by PawelGutkowski** [comment]
@mogoodrich I've adressed your comments.

**Comment by mogoodrich** [comment]
Thanks @PawelGutkowski, I will merge shortly!

**Pull request merged**